File: backend/BlogPostApi/Models/BlogPost.cs
namespace BlogPostApi.Models
{
    public class BlogPost
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string Author { get; set; } = string.Empty;
        public int UserId { get; set; } // Foreign key
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int FlagCount { get; set; } = 0;
        
        // Navigation properties
        public User User { get; set; } = null!;
    }
}

File: frontend/src/types/BlogPost.ts
import type { User } from './Auth';

export interface BlogPost {
  id: number;
  title: string;
  content: string;
  author: string;
  userId: number;
  createdAt: string;
  updatedAt: string;
  isOwner: boolean;
  user: User;
}

export interface CreateBlogPost {
  title: string;
  content: string;
}

export interface UpdateBlogPost {
  title: string;
  content: string;
}